src/mscorlib/src/System/Array.cs

```diff
@@ -1011,6 +1011,20 @@ public Object Clone()
             return ArraySortHelper<T>.Default.BinarySearch(array, index, length, value, comparer);
         }
 
+        public static T[] Clone<T>(T[] array)
+        {
+            if (array == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
+            }
+
+            // Note: We should always return a new instance, even for empty arrays.
```

Comment:
I suggest you include in the comment why it's important not to return a new instance.  That way, in the future we can reevaluate, factoring in the original reasoning that led to this, and choose to change it.  There are other places where Clone methods on immutable objects return ```this```, e.g. string.Clone, and an empty array is an immutable object.